integration-tests-support/kafka/src/main/java/org/apache/camel/quarkus/test/support/kafka/KafkaTestResource.java

```diff
@@ -18,38 +18,96 @@
 
     private StrimziKafkaContainer container;
+    private GenericContainer j17container;
 
     @Override
     public Map<String, String> start() {
         LOGGER.info(TestcontainersConfiguration.getInstance().toString());
 
         try {
-            container = new StrimziKafkaContainer(KAFKA_IMAGE_NAME)
-                    /* Added container startup logging because of https://github.com/apache/camel-quarkus/issues/2461 */
-                    .withLogConsumer(frame -> System.out.print(frame.getUtf8String()))
-                    .waitForRunning();
-
-            container.start();
+            startContainer(KAFKA_IMAGE_NAME, name -> new StrimziKafkaContainer(name));
 
             return Collections.singletonMap("camel.component.kafka.brokers", container.getBootstrapServers());
         } catch (Exception e) {
             throw new RuntimeException(e);
         }
     }
 
+    public String start(Function<String, StrimziKafkaContainer> containerSupplier) {
+        LOGGER.info(TestcontainersConfiguration.getInstance().toString());
+
+        //if FIPS environment is present, custom container using J17 has to used because:
+        // Password-based encryption support in FIPs mode was implemented in the Red Hat build of OpenJDK 17 update 4
+        if (FipsModeUtil.isFipsMode()) {
+            //custom image should be cached for the next usages with following id
+            String customImageName = "camel-quarkus-test-custom-" + KAFKA_IMAGE_NAME.replaceAll("[\\./]", "-");
+
+            try {
+                //in case that the image is not accessible, fetch exception is thrown
+                startContainer(customImageName, containerSupplier);
+            } catch (ContainerFetchException e) {
+                if (e.getCause() instanceof NotFoundException) {
+                    LOGGER.info("Custom image for kafka (%s) does not exist. Has to be created.", customImageName);
+
+                    //start of the customized container will create the image
+                    //it is not possible to customize existing StrimziKafkaContainer. Testcontainer API doe not allow
+                    //to customize the image.
+                    // This workaround can be removed once the strimzi container with openjdk 17 is released.
+                    // According to https://strimzi.io/blog/2023/01/25/running-apache-kafka-on-fips-enabled-kubernetes-cluster/
+                    // image should exist
+                    j17container = new GenericContainer(
+                            new ImageFromDockerfile(customImageName, false)
+                                    .withDockerfileFromBuilder(builder -> builder
+                                            .from("quay.io/strimzi-test-container/test-container:latest-kafka-3.2.1")
+                                            .env("JAVA_HOME", "/usr/lib/jvm/jre-17")
+                                            .env("PATH", "/usr/lib/jvm/jre-17/bin:$PATH")
```

Comment:
Any idea if there's any open issues with Strimzi to see if they can switch to JDK 17 in their container images?

If not, we should maybe create one.

Reply:
I found this blog post, which indicates, that the issue was fixed - https://strimzi.io/blog/2023/01/25/running-apache-kafka-on-fips-enabled-kubernetes-cluster/
But I tried the newest image and the java is still 11. So I'm not sure what is wrong.

Reply:
Here's the reason I guess:

https://github.com/strimzi/test-container-images/blob/main/images/base/Dockerfile#L6

Reply:
I'll report the issue to the strimzi and propose a fix. We can remove workaround once the issue is fixed

Reply:
Thanks for the link!

Reply:
raised a ticket https://github.com/strimzi/test-container-images/issues/25 and proposed a fix https://github.com/strimzi/test-container-images/pull/26.
Here is a ticket for removing the workaround https://github.com/apache/camel-quarkus/issues/6097